plonk-core/src/proof_system/proof.rs

```diff
@@ -117,6 +117,10 @@ where
                 adicity: <<F as FftField>::FftParams as ark_ff::FftParameters>::TWO_ADICITY,
             })?;
 
+        for val in pub_inputs {
+            transcript.append(b"pi", val)
+        }
```

Comment:
Are the `pi` recieved sparse?? 
If so, we should see if we can avoid adding the zeros and filter them out.

Reply:
They usually are, so now only non-zero values will be appended to the transcript.